src/Symfony/Bundle/DebugBundle/Resources/config/services.xml

```diff
@@ -20,19 +22,19 @@
             <argument>null</argument><!-- var_dumper.cli_dumper when debug.dump_destination is set -->
         </service>
 
-        <service id="debug.dump_listener" class="Symfony\Component\HttpKernel\EventListener\DumpListener">
+        <service id="debug.dump_listener" class="Symfony\Component\HttpKernel\EventListener\DumpListener" public="true">
```

Comment:
The form types I linked are the one that are not deprecated because they need special instantiation.
You're right that this trick is valid only on 3.4. But maybe 3.4 is the last one that will have public services? So if using the trick is enough for 3.4, then maybe better not to work on a new feature for nothing?

Anyway, that's an interesting discussion, but should not be hidden in this PR :)